js/geo/transform.js

```diff
@@ -375,7 +394,8 @@ Transform.prototype = {
                 y2 !== undefined ? y2 : this.y));
         }
 
-        this._constraining = false;
+        this._unmodified = this._constraining.unmodified;
+        this._constraining = undefined;
```

Comment:
We can't store it in a local variable because `_constrain()` recurses.  

I think it's pretty clear that I'm just preserve the `_unmodified` state between recursive calls.  I'm surprised that you think this code "looks a bit weird" but I'm happy to add some comments to make it a bit more obvious what's going on if that's what it would take to move forward on this.